gnome-initial-setup/gnome-initial-setup-copy-worker.c

```diff
@@ -69,18 +71,47 @@ move_file_from_homedir (GFile       *src_base,
 static void
 unlock_keyring (const gchar *initial_setup_homedir)
 {
-  gchar *file;
-  gchar *password = NULL;
+  g_autofree gchar *file = g_build_filename (initial_setup_homedir, ".config/password", NULL);
+  gsize password_size;
+  g_autofree gchar *password = NULL;
+  SecretValue *secret = NULL;
+  SecretService *service = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (!g_file_get_contents (file, &password, &password_size, &error)) {
+    g_warning ("Unable to read user password file: %s", error->message);
+    goto out;
+  }
 
-  file = g_build_filename (initial_setup_homedir, ".config/password", NULL);
-  if (g_file_get_contents (file, &password, NULL, NULL))
-    gnome_keyring_unlock_sync ("login", password);
-  else
-    g_warning ("Unable to read user password file");
+  service = secret_service_get_sync (SECRET_SERVICE_OPEN_SESSION, NULL, &error);
+  if (service == NULL) {
+    g_warning ("Failed to get secret service: %s", error->message);
+    goto out;
+  }
+
+  secret = secret_value_new (password, password_size, "text/plain");
+
+  g_dbus_connection_call_sync (g_dbus_proxy_get_connection (G_DBUS_PROXY (service)),
+                               "org.gnome.keyring",
```

Comment:
Presumably you’re using the well-known name `org.gnome.keyring` rather than `org.freedesktop.secrets`, because you can’t guarantee that `org.gnome.keyring.InternalUnsupportedGuiltRiddenInterface` is not guaranteed to be exposed on `org.freedesktop.secrets` (if it was implemented by the KDE daemon, for example)?

Might be worth a comment to that effect.

Reply:
Might also want to add a runtime dependency on `gnome-keyring` in `debian/control` too, for completeness.

Reply:
For the record I'm using the same well-known name as the existing upstream code in `gis-keyring.c` does; yes, presumably that reason.

Dependency – I suppose for completeness one should do that but I don't think it's worth a downstream patch. I'll send a patch to Debian.

Reply:
https://salsa.debian.org/gnome-team/gnome-initial-setup/merge_requests/4